pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -57,38 +57,38 @@ func NewWatcher(sockDir string) Watcher {
 }
 
 // AddHandler registers a callback to be invoked for a particular type of plugin
-func (w *Watcher) AddHandler(handlerType string, handlerCbkFn RegisterCallbackFn) {
+func (w *Watcher) AddHandler(pluginType string, handlerCbkFn RegisterCallbackFn) {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
-	w.handlers[handlerType] = handlerCbkFn
+	w.handlers[pluginType] = handlerCbkFn
 }
 
 // Creates the plugin directory, if it doesn't already exist.
 func (w *Watcher) createPluginDir() error {
 	glog.V(4).Infof("Ensuring Plugin directory at %s ", w.path)
 	if err := w.fs.MkdirAll(w.path, 0755); err != nil {
-		return fmt.Errorf("error (re-)creating driver directory: %s", err)
+		return fmt.Errorf("error (re-)creating root %s: %s", w.path, err)
 	}
+
 	return nil
 }
 
-// Walks through the plugin directory to discover any existing plugin sockets.
-func (w *Watcher) traversePluginDir() error {
-	files, err := w.fs.ReadDir(w.path)
+// Walks through the plugin directory discover any existing plugin sockets.
+func (w *Watcher) traversePluginDir(dir string) error {
+	files, err := w.fs.ReadDir(dir)
```

Comment:
I think the use `filepath.Walk` to traverse the path would work better and probably make logic simpler.

Reply:
been addressed, thanks